plugin/trino-phoenix5/src/main/java/io/trino/plugin/phoenix5/PhoenixClient.java

```diff
@@ -265,6 +265,60 @@ public ResultSet executeQuery()
         };
     }
 
+    public PreparedStatement prepareStatement(
+            ConnectorSession session,
+            Connection connection,
+            JdbcTableHandle table,
+            List<JdbcColumnHandle> columns,
+            Optional<JdbcSplit> split)
+            throws SQLException
+    {
+        PreparedQuery preparedQuery = prepareQuery(
+                session,
+                connection,
+                table,
+                Optional.empty(),
+                columns,
+                ImmutableMap.of(),
+                split);
+        return new QueryBuilder(this).prepareStatement(session, connection, preparedQuery);
+    }
+
+    @Override
+    public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, List<JdbcSortItem> sortOrder)
+    {
+        return true;
+    }
+
+    @Override
+    protected Optional<TopNFunction> topNFunction()
+    {
+        return Optional.of((query, sortItems, limit) -> {
+            // TODO: Remove when this is fixed in Phoenix
+            // Phoenix does deal well with large LIMITs with ORDER BY
```

Comment:
`s/does/doesn't/`

Reply:
I need to fix the comment anyway. I fixed the Phoenix bug that over-estimated the required memory. Now it's because for large values here Trino is just faster. (Looking into that as well, but as is, it's better to do this in Trino).

Reply:
Doe